sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java

```diff
@@ -2207,12 +2234,62 @@ public VirtualMachineImpl withAvailabilityZone(AvailabilityZoneId zoneId) {
         return this;
     }
 
+    @Override
+    public VirtualMachineImpl withOsDiskDeleteOptions(DeleteOptions deleteOptions) {
+        if (deleteOptions == null
+            || this.innerModel().storageProfile() == null || this.innerModel().storageProfile().osDisk() == null) {
+            return null;
+        }
+        this.innerModel().storageProfile().osDisk().withDeleteOption(diskDeleteOptionsFromDeleteOptions(deleteOptions));
+        return this;
+    }
+
     @Override
     public VirtualMachineImpl withPrimaryNetworkInterfaceDeleteOptions(DeleteOptions deleteOptions) {
         this.primaryNetworkInterfaceDeleteOptions = deleteOptions;
         return this;
     }
 
+    @Override
+    public VirtualMachineImpl withNetworkInterfacesDeleteOptions(DeleteOptions deleteOptions, String... nicIds) {
+        if (nicIds == null || nicIds.length == 0) {
+            throw new IllegalArgumentException("No nicIds specified for `withNetworkInterfacesDeleteOptions`");
+        }
+        if (this.innerModel().networkProfile() != null
+            && this.innerModel().networkProfile().networkInterfaces() != null) {
+            Set<String> nicIdSet = new HashSet<>(Arrays.asList(nicIds));
+            this.innerModel().networkProfile().networkInterfaces().forEach(
+                nic -> {
+                    if (nicIdSet.contains(nic.id())) {
```

Comment:
Should it do "compare ignore case" on ID?

https://github.com/Azure/azure-sdk-for-java/blob/676648838bd445bf27090fddb3845a68345dbe00/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java#L1430C29-L1430C45

Reply:
Yeah, good catch. Resource names are case-insensitive.
> When using various APIs to retrieve the name for a resource or resource group, the returned value may have different casing than what you originally specified for the name. The returned value may even display different case values than what is listed in the valid characters table.